src/libstd/macros.rs

```diff
@@ -663,3 +663,39 @@ pub mod builtin {
     #[macro_export]
     macro_rules! include { ($file:expr) => ({ /* compiler built-in */ }) }
 }
+
+/// A macro for defining #[cfg] if-else statements.
+///
+/// This is similar to the `if/elif` C preprocessor macro by allowing definition
+/// of a cascade of `#[cfg]` cases, emitting the implementation which matches
+/// first.
+///
+/// This allows you to conveniently provide a long list #[cfg]'d blocks of code
+/// without having to rewrite each clause multiple times.
+macro_rules! cfg_if {
```

Comment:
I suppose there's a reason we can't bring in [cgf_if](https://crates.io/crates/cfg_if)?

Reply:
Unfortunately it's not so easy to pull in crates as a dep of libstd, and its a tiny macro, so I didn't mind duplicating it in a few locations